drivers/mtd_flashpage/mtd_flashpage.c

```diff
@@ -40,6 +49,8 @@ static int _init(mtd_dev_t *dev)
 
 static int _read(mtd_dev_t *dev, void *buf, uint32_t addr, uint32_t size)
 {
+    addr = _mtd_to_flashpage_addr(addr);
```

Comment:
I'd say it's better not to reuse `addr` here. Please change it to something like this:

```suggestion
    uword_t addr_abs = _mtd_to_flashpage_addr(addr);
```

And use `addr_abs` in the implementation